api/src/domain/tax-filings/taxFilingsInterfaceFactory.ts

```diff
@@ -23,36 +23,54 @@ export const taxFilingsInterfaceFactory = (apiTaxFilingClient: TaxFilingClient):
   };
 
   const onboarding = async (request: taxFilingInterfaceRequest): Promise<UserData> => {
-    const userData = await lookup(request);
-    if (userData.taxFilingData.state === "UNREGISTERED") {
-      const response = await apiTaxFilingClient.onboarding(
-        request.taxId,
-        userData.user.email,
-        request.businessName
-      );
-      const registered = ["SUCCESS", "PENDING"].includes(response.state);
-      return {
-        ...userData,
-        taxFilingData: {
-          ...userData.taxFilingData,
-          businessName: request.businessName,
-          lastUpdatedISO: new Date(Date.now()).toISOString(),
-          registeredISO: registered ? new Date(Date.now()).toISOString() : undefined,
-          errorField: response.errorField,
-          state: response.state == "SUCCESS" ? "PENDING" : response.state,
-        },
-      };
-    } else {
-      const registered = ["SUCCESS", "PENDING"].includes(userData.taxFilingData.state ?? "");
-      return {
-        ...userData,
-        taxFilingData: {
-          ...userData.taxFilingData,
-          registeredISO:
-            userData.taxFilingData.registeredISO ??
-            (registered ? new Date(Date.now()).toISOString() : undefined),
-        },
-      };
+    const response = await apiTaxFilingClient.onboarding(
+      request.taxId,
+      request.userData.user.email,
+      request.businessName
+    );
+
+    switch (response.state) {
+      case "FAILED": {
+        return {
+          ...request.userData,
+          taxFilingData: {
+            ...request.userData.taxFilingData,
+            registeredISO: undefined,
+            state: response.state,
+            errorField: response.errorField,
+            businessName: request.businessName,
+          },
+        };
+      }
+      case "PENDING": {
+        return {
+          ...request.userData,
+          taxFilingData: {
+            ...request.userData.taxFilingData,
+            lastUpdatedISO: new Date(Date.now()).toISOString(),
+            registeredISO: new Date(Date.now()).toISOString(),
+            errorField: response.errorField,
+            state: response.state,
+            businessName: request.businessName,
+          },
+        };
+      }
+      case "SUCCESS": {
+        return await lookup(request);
+      }
+
+      default: {
```

Comment:
I notice this is the same as the "FAILED" case, right? and also that API_ERROR case doesn't have its own specific case, so I assume it's meant to be covered by the default?

A cleaner way to write this could be to use the fall-through feature of switch statements to group shared behavior:

```
case "API_ERROR":
case "FAILED":
default:
  return ... 
```

This a) groups shared behavior to reduce duplication and b) makes it clear all the possible cases we know about, so that someone reading it doesn't think that API_ERROR has just been forgotten. Instead, it acknowledges that its a known case, and explicitly states its outcome